### PR TITLE
Checking sizes of input arrays

### PR DESCRIPTION
**Pull request by abhinaukumar**
I have added code to check the shapes of the b1, gr, and tp variables in bloch/bloch.py. The shapes of the other two arrays seem to be independent of these variables, and each other, so I haven't checked their shapes.

### CODE REVIEW

**Comment by jtamir** [comment]
Thanks! Did you run the tests?

File: bloch/bloch.py [hunk 1]
```diff
@@ -43,7 +43,20 @@ def bloch(b1, gr, tp, t1, t2, df, dp, mode, mx=None, my=None, mz=None):
...
+
+    assert gr.ndim == 2 and gr.shape[0] <= 3 and gr.shape[1] == ntime, 'gr is of the wrong shape.'
+
+    tp_flag = False
+    tp_flag |= isinstance(tp, NUMBER)
+    tp_flag |= (tp.ndim == 1)
+    if tp.ndim == 2:
+        tp_flag |= (tp.shape[0] == 1) and ((tp.shape[1] == 1) or (tp.shape[1] == ntime))
+    else:
+        tp_flag |= False
```
Comment (jtamir) [inline bloch/bloch.py]: Is this necessary?

Comment (abhinaukumar) [reply bloch/bloch.py]: Technically, no. I just added it because I thought it might be easier to read. I'll remove it.

**Comment by abhinaukumar** [comment]
I forgot, sorry. I ran the tests now and they're failing, which actually brings me to a question that I wanted to ask you. When you get a scalar input for `b1` (`NUMBER` type), you "promote" it to a 1x1 array and interpret it as a simulation for one time-step. I think there is ambiguity when we try to promote the gradient array. Suppose we get an array of size (3,). Do we want to interpret it as a (1,3) array or a (3,1) array? Beyond 3, it's clear that we should resize (N,) into (1, N) because the first dimension can be of size at most 3, but for N <= 3, it seemed a little ambiguous to me. So, I was forcing the gradient array to explicitly be 2-dimensional to avoid having to promote it, which is why the tests fail.

**Comment by jtamir** [comment]
We don't want to break backwards compatibility. Can you change it to interpret the old way for array of size `(N,)` where `N <= 3` and make a warning about ambiguity and advising to explicitly make it 2d?

**Comment by abhinaukumar** [comment]
I should've read the code better. The routines in bloch_processing.py were already dedicated to preprocessing inputs. So, I moved the input checks into those functions and wrote an extra routine for RF. I also added a warning about small 1-dim gradient arrays, but I haven't changed the way it was being treated, so it will be backwards compatible. All the unit tests passed, but there is now one warning because one of the inputs is a 1-dim gradient array of size (3,). This is the desired behaviour, so nothing to worry about.

File: bloch/bloch.py (hunk 1 above)
Comment (abhinaukumar) [reply bloch/bloch.py]: This is no longer a concern after 0aa5b7b.

**Comment by jtamir** [comment]
Looks great! Thanks for putting this together.

**Comment by abhinaukumar** [comment]
No problem! I'm happy to help in any small way.

**Pull request merged**